internal/handler/user_service.go

```diff
@@ -0,0 +1,101 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/gobwas/glob"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/smailic05/TaskInfoblox/internal/pb"
+)
+
+type User struct {
+	Address  string
+	Username string
+	Phone    string
+}
+
+type UserService struct {
+	pb.UnimplementedUserServiceServer
+	UserSlice []User
+}
+
+func New() *UserService {
+	return &UserService{UserSlice: make([]User, 0)}
+}
+
+func (s *UserService) AddUser(ctx context.Context, addUser *pb.AddUserRequest) (*pb.AddUserResponse, error) {
+	user := User{
+		Address:  addUser.Address,
+		Username: addUser.Username,
+		Phone:    addUser.Phone}
+	s.UserSlice = append(s.UserSlice, user)
+	return &pb.AddUserResponse{Response: "Success"}, nil
+}
+
+func (s *UserService) DeleteUser(ctx context.Context, deleteUser *pb.DeleteUserRequest) (*pb.DeleteUserResponse, error) {
+	name, err := glob.Compile(deleteUser.Username)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Error")
+	}
+	address, err := glob.Compile(deleteUser.Address)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Error")
+	}
+	phone, err := glob.Compile(deleteUser.Phone)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "Error")
+	}
+
```

Comment:
Can we move common code of checking request to a separate function?

I see 2 points of using